lib/ConvoyLeader/src/DrivingState.cpp

```diff
@@ -73,12 +73,24 @@ void DrivingState::process(StateMachine& sm)
     }
     else
     {
+        int16_t maxPossibleSpeed = m_maxMotorSpeed;
+
         /* TODO: Check limits. */
 
         /* TODO: Check follower feedback. Calculate platoon length and react accordingly */
 
+        /* Limit speed based on the proximity sensors. */
+        /* m_vehicleData.proximity goes from 0 to NUM_PROXIMITY_SENSOR_RANGES */
+        maxPossibleSpeed = m_maxMotorSpeed - (m_vehicleData.proximity * m_rangeFactor);

```

Comment:
Is there any processing time considered, e.g. from detection time to reaction time 10 ms or more? Or just a proof-of-concept?